app/controllers/saml_idp_controller.rb

```diff
@@ -57,7 +58,12 @@ def store_location_and_redirect_to(url)
 
   def render_template_for(message, action_url, type)
     domain = SecureHeadersWhitelister.extract_domain(action_url)
-    override_content_security_policy_directives(form_action: ["'self'", domain])
+    csp_uris = ["'self'", domain]
+
+    additional_csp_uris = decorated_session.sp_redirect_uris || []
+    csp_uris |= additional_csp_uris.compact unless additional_csp_uris.empty?
+
```

Comment:
What do you think about moving this logic into a new method in `SecureHeadersWhitelister`?

Reply:
I think that would make sense, I'm extracting out the logic and updating the PR momentarily.